src/linear_algebra/bilinear_form.lean

```diff
@@ -351,117 +391,322 @@ end
 
 end
 
+section is_basis
+
+variables {B₃ F₃ : bilin_form R₃ M₃}
+variables {ι : Type*} {b : ι → M₃} (hb : is_basis R₃ b)
+
+/-- Two bilinear forms are equal when they are equal on all basis vectors. -/
+lemma ext_basis (h : ∀ i j, B₃ (b i) (b j) = F₃ (b i) (b j)) : B₃ = F₃ :=
+to_lin.injective (hb.ext (λ i, hb.ext (λ j, h i j)))
```

Comment:
could use dollars here, but up to you
```suggestion
to_lin.injective $ hb.ext $ λ i, hb.ext $ λ j, h i j
```

Reply:
Or I suppose `to_lin.injective $ hb.ext $ hb.ext ∘ h` but that's probably madness

Reply:
Maybe not madness, definitely pointless :)